src/hooks/useDragAndDrop.ts

```diff
@@ -82,23 +74,23 @@ export default function useDragAndDrop({dropZone, onDrop = () => {}, shouldAllow
                     handleDragEvent(event);
                     break;
                 case DRAG_ENTER_EVENT:
-                    dragCounter.current++;
                     handleDragEvent(event);
                     if (isDraggingOver) {
                         return;
                     }
                     setIsDraggingOver(true);
                     break;
                 case DRAG_LEAVE_EVENT:
-                    dragCounter.current--;
-                    if (!isDraggingOver || dragCounter.current > 0) {
+                    // This is necessary because dragging over children will cause dragleave to execute on the parent.
+                    const currentTarget = event.currentTarget as HTMLElement | null;
+                    const relatedTarget = event.relatedTarget as HTMLElement | null;
+                    if (!isDraggingOver || currentTarget?.contains(relatedTarget)) {
                         return;
                     }
```

Comment:
```suggestion
                    if (!isDraggingOver) {
                        return;
                    }

                    // This is necessary because dragging over children will cause dragleave to execute on the parent.
                    if ((event.currentTarget as HTMLElement).contains(event.relatedTarget as HTMLElement)) {
                        return;
                    }
```

Reply:
```js
if ((event.currentTarget as HTMLElement | null)?.contains(event.relatedTarget as HTMLElement | null)) {
```

Right. i think this is safer